hack/test-end-to-end.sh

```diff
@@ -379,6 +379,37 @@ echo "[INFO] Validating port-forward"
 oc port-forward -p ${registry_pod} 5001:5000  &> "${LOG_DIR}/port-forward.log" &
 wait_for_url_timed "http://localhost:5001/healthz" "[INFO] Docker registry says: " $((10*TIME_SEC))
 
+# Image pruning
+echo "[INFO] Validating image pruning"
+# get a 2nd image to work with
+docker pull centos:centos7
+
+# tag and push 1st image - layers unique to this image will be pruned
+docker tag -f openshift/ruby-20-centos7:latest ${DOCKER_REGISTRY}/cache/prune
+docker push ${DOCKER_REGISTRY}/cache/prune
+
+# tag and push 2nd image - it won't be pruned
+docker tag -f centos:centos7 ${DOCKER_REGISTRY}/cache/prune
+docker push ${DOCKER_REGISTRY}/cache/prune
+
+# record the storage before pruning
+oc exec -p ${registry_pod} du /registry > ${LOG_DIR}/prune-images.before.txt
+
+# set up pruner user
+oadm policy add-cluster-role-to-user system:image-pruner e2e-pruner
+oc login -u e2e-pruner -p pass
+
+# run image pruning
+oadm prune images --keep-younger-than=0 --keep-tag-revisions=1 --confirm &> ${LOG_DIR}/prune-images.log
+! grep error ${LOG_DIR}/prune-images.log
+
+oc project ${CLUSTER_ADMIN_CONTEXT}
+# record the storage after pruning
+oc exec -p ${registry_pod} du /registry > ${LOG_DIR}/prune-images.after.txt
+
+# make sure there were changes to the registry's storage
+[ -n "$(diff ${LOG_DIR}/prune-images.before.txt ${LOG_DIR}/prune-images.after.txt)" ]
+
```

Comment:
I don't mind doing this, but it's not consistent with the rest of this file. test-cmd does echoes like this, but not e2e (yet).